### PR TITLE
STRF-6701 Renderer: reduce usage of Lodash

### PR DESCRIPTION
**Pull request by bc-williamkwon**
## What? Why?
STRF-6701 Renderer: Reduce usage of Lodash.  Lodash utility methods can be slow especially things like _.each and this is a pr to improve performance of Storefront Renderer by moving to native JavaScript methods and for loops.

## How was it tested?

----
Unit tests still pass.
cc @bigcommerce/storefront-team

### CODE REVIEW

File: helpers/getImage.js [hunk 1]
```diff
@@ -1,23 +1,22 @@
...
-        var presets = settings._images;
-        var size;
-        var width;
-        var height;
-
-        if (!_.isPlainObject(image) || !_.isString(image.data) || image.data.indexOf('{:size}') === -1) {
+        const sizeRegex = /^(\d+?)x(\d+?)$/g;
+        const settings = globals.getThemeSettings() || {};
+        const presets = settings._images;
+        let size;
```
Comment (mattolson) [inline helpers/getImage.js]: Prefer one line for this: `let size, width, height;`

File: helpers/lib/common.js [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+        return !!val.length;
+    } else if (isEmptyObject(val)) {
+        return false;
+    } else {
+        return !!val;
+    }
+}
+//function to check if two values are equal also does a check for NaN === NaN
+function equals(val1, val2) {
+    return val1 === val2 || (val1 !== val1 && val2 !== val2);
```
Comment (mattolson) [inline helpers/lib/common.js]: Can we use https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Object/is for this?

Comment (bc-williamkwon) [reply helpers/lib/common.js]: Learn something new everyday.  I believe we can.

File: helpers/getImage.js (hunk 1 above)
Comment (bc-williamkwon) [reply helpers/getImage.js]: ok, changed.

File: helpers/lib/common.js (hunk 2 above)
Comment (bc-williamkwon) [reply helpers/lib/common.js]: fixed.

File: helpers/lib/common.js [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+    if (Array.isArray(val)) {
+        return !!val.length;
+    } else if (isEmptyObject(val)) {
+        return false;
+    } else {
+        return !!val;
+    }
+}
+//function to check if two values are equal also does a check for NaN === NaN
+function equals(val1, val2) {
```
Comment (mattolson) [inline helpers/lib/common.js]: I think now that we have an equivalence for this, we really don't need this method anymore, let's just use `Object.is` whenever we need that behavior. I don't see a need for it in the current PR.

Comment (bc-williamkwon) [reply helpers/lib/common.js]: I have removed the function.

**Pull request merged**